src/scan.ts

```diff
@@ -55,13 +56,25 @@ export async function scan(scanOptions: ScanOptions, cliArgs?: string[]) {
     `JRE Provisioning ${supportsJREProvisioning ? 'is ' : 'is NOT '}supported on ${serverUrl}`,
   );
 
+  if (!supportsJREProvisioning) {
+    log(LogLevel.INFO, 'Will download and use sonar-scanner-cli');
+    if (scanOptions.localScannerCli) {
+      log(LogLevel.INFO, 'Local scanner is requested, will not download sonar-scanner-cli');
+      if (!(await tryLocalSonarScannerExecutable(SCANNER_CLI_DEFAULT_BIN_NAME))) {
+        throw new Error('Local scanner is requested but not found');
+      }
+      await runScannerCli(scanOptions, properties, SCANNER_CLI_DEFAULT_BIN_NAME);
+    } else {
+      const binPath = await downloadScannerCli(properties);
+      await runScannerCli(scanOptions, properties, binPath);
```

Comment:
https://github.com/SonarSource/sonar-scanner-npm/blob/master/src/sonar-scanner-executable.js#L125

there's a TODO to check the version is at least 2.8. @vdiez any idea why this is needed?

Reply:
it's a 7-year-old comment; probably Fabrice has an explanation. IMHO, we could just mention that in the README